integration_test/update_test.go

```diff
@@ -34,3 +47,62 @@ func TestKrewUpdate(t *testing.T) {
 		t.Fatal("re-run of 'update' must succeed")
 	}
 }
+
+func TestKrewUpdateListsNewPlugins(t *testing.T) {
+	skipShort(t)
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test = test.WithIndex()
+	if err := os.Remove(filepath.Join(test.Root(), "index", "plugins", validPlugin+constants.ManifestExtension)); err != nil {
```

Comment:
Better use `environment.NewPaths(test.Root()).IndexPluginsPath()`.

Same below

Reply:
Normally I'd say this is valid. But tests ideally should rely on these utilities a lot less, as that's sort of what they're testing.

In this case, I guess we can use the utility as that's not what we're testing.